integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -246,6 +247,126 @@ object BenchUtils {
     }
   }
 
+  /**
+   * Perform a diff of the results collected from two DataFrames, allowing for differences in
+   * precision.
+   *
+   * This is only suitable for data sets that can fit in the driver's memory.
+
```

Comment:
Could we also add in a note about how this could add in a sort and does a collect so for benchmark results to be accurate in isolation it is probably best to run the benchmark with the output going to a file and this compares the results of that output after timing has stopped, or that this is used to compare results for a single run after more timings have been collected for other runs.